packages/qdrant-loader-mcp-server/src/qdrant_loader_mcp_server/search/nlp/semantic_expander.py

```
@@ -176,200 +168,216 @@ def expand_query(
-                    len(token.text) > 2 and
-                    token.pos_ in {"NOUN", "VERB", "ADJ"})
+                token.lemma_.lower()
+                for token in doc
+                if (
+                    token.is_alpha
+                    and not token.is_stop
+                    and len(token.text) > 2
+                    and token.pos_ in {"NOUN", "VERB", "ADJ"}
+                )
             ]
         except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)